crates/core/src/db/queries/featured_listings.rs

```diff
@@ -0,0 +1,66 @@
+//! Query utilities for featured listings.
+
+use anyhow::Context;
+use diesel::{
+    pg::Pg,
+    sql_types::{Array, Integer, Nullable, Text},
+    types::ToSql,
+    RunQueryDsl,
+};
+
+use crate::{
+    db::{models::ListingReceipt, Connection},
+    error::Result,
+};
+
+const FEATURED_LISTINGS_QUERY: &str = r"
+SELECT a.address, a.trade_state, a.bookkeeper, a.auction_house, a.seller, a.metadata, a.purchase_receipt, a.price, a.token_size, a.bump, a.trade_state_bump, a.created_at, a.canceled_at from (
+    SELECT listing_receipts.*, row_number() OVER (PARTITION BY listing_receipts.seller ORDER BY listing_receipts.price DESC) as row
```

Comment:
Looks like this exceeds the formatter's 100-char line limit, can we get some newlines?